src/inference/src/dev/core_impl.cpp

```diff
@@ -734,13 +735,20 @@ void ov::CoreImpl::set_property_for_device(const ov::AnyMap& configMap, const st
         std::lock_guard<std::mutex> lock(get_mutex());
         created_plugins.reserve(plugins.size());
 
+        // TODO: keep only:
+        //    coreConfig.set_and_update(config);
+        // once GPU remove support of ov::cache_dir
+        // CoreConfg::set_and_update will drop CACHE_DIR from config map
+        // and updates core config with new ov::cache_dir
```

Comment:
I supposed that ov::cache_dir will be removed from all plugin and only exist on ov::core level, it is reasonable.
If so, we also need PR to remove ov::cache_dir from meta plugin(AUTO/BATCH/HETERO)

Reply:
good point that meta plugins support cache_dir to transfer it to underlying HW plugins.
Need to think how to simplify this handling.

In general, our goal is to avoid handling of core properties by META plugins to simplify them.